index.js

```diff
@@ -267,6 +289,62 @@ class Rook extends Piece {
   }
 }
 
+class Bishop extends Piece {
+  constructor(game, board, color, row, file) {
+    super(game, board, color);
+    this.row = row;
+    this.file = file;
+  }
+
+  name = "Bishop";
+
+  nearestSquareOccupied(square) {
+    if (this.isSquareOccupied(...square)) {
+      return square;
+    }
+  }
+
+  moveDiagonally(row, file, arr, vert, horiz) {
+    const upDown = vert === "up" ? 1 : -1
+    const leftRight = horiz === "right" ? 1 : -1
+
+    for (let i = 1; i < Board.LANE_SIZE; i++) {
+      const square = [row + i * upDown, file + i * leftRight]
+      if (this.isValidSquare(...square) && !this.isSquareOccupied(...square)) {
+        arr.push(Move.fromSquare(square, this));
+      } else if (
+        this.isValidSquare(...square) &&
+        this.isSquareOccupied(...square)
+      ) {
+        if (this.getSquareContent(...square).color !== this.color) {
+          arr.push(
+            Move.fromSquare(square, this, this.getSquareContent(...square))
+          );
+        }
+        return arr;
+      }
+    }
+    return arr
+  }
+
+  get moves() {
+    const available = []
+
+    this.moveDiagonally(this.row, this.file, available, "up", "right");
+    this.moveDiagonally(this.row, this.file, available, "down", "left");
+    this.moveDiagonally(this.row, this.file, available, "up", "left");
+    this.moveDiagonally(this.row, this.file, available, "down", "right");
+    return available
+  }
```

Comment:
This is getting really repetitive. Let's extract a top-level function called `getLegalDirectionalMoves` that takes as input `(piece: Piece, directions: [number, number])` (to be read as "A piece of type Piece, and a direction of type Array with two Numbers") and returns an array of available moves. If you've done this correctly, the above code will become

```suggestion
  get moves() {
    const directions = [[-1, -1], [-1, 1], [1, -1], [1, 1]]
    const available = directions.flatMap(dir => getLegalDirectionalMoves(this, direction))
    return available
  }
```